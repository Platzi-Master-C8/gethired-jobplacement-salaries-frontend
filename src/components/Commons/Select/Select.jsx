import React from 'react';
import PropTypes from 'prop-types';

import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import FormHelperText from '@mui/material/FormHelperText';
import InputLabel from '@mui/material/InputLabel';
import SelectMUI from '@mui/material/Select';

const Select = (props) => {
    const { helperText, options, id, label, value, onChange, multiple, disabled, name, ...otherProps } = props;
    return (
        <FormControl variant="filled" sx={{ my: 1 }} fullWidth disabled={disabled}>
            <InputLabel id={id}>{label}</InputLabel>
            <SelectMUI labelId={id} value={value} onChange={onChange} multiple={multiple} name={name} {...otherProps}>
                {options?.map((option) => (
                    <MenuItem key={option} value={option}>
                        {option}
                    </MenuItem>
                ))}
            </SelectMUI>
            {!!helperText && <FormHelperText>{helperText}</FormHelperText>}
        </FormControl>
    );
};

Select.propTypes = {
    onChange: PropTypes.func.isRequired,
    multiple: PropTypes.bool,
    name: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    id: PropTypes.string.isRequired,
    helperText: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.string).isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]).isRequired,
};

Select.defaultProps = {
    helperText: '',
    multiple: false,
    disabled: false,
};

export default Select;
