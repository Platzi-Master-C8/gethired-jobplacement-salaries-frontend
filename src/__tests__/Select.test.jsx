import React from 'react';
import { render, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Select from 'Components/Commons/Select';

describe('Tests for component <Select />', () => {
    const handleChange = jest.fn();
    const props = {
        value: '',
        onChange: handleChange,
        id: 'label-select',
        label: 'Select option',
        name: 'select',
        options: ['test 1', 'test 2'],
    };

    test('Take Snapshot', () => {
        const { asFragment } = render(<Select {...props} />);

        expect(asFragment()).toMatchSnapshot();
    });
    test('Select with label Select option and click', async () => {
        const { findByLabelText, findByRole } = render(<Select {...props} />);
        const selectLabel = /Select option/i;
        const selectEl = await findByLabelText(selectLabel);

        expect(selectEl).toBeInTheDocument();

        userEvent.click(selectEl);

        const optionsPopupEl = await findByRole('listbox');

        userEvent.click(within(optionsPopupEl).getByText(/test 1/i));

        expect(handleChange).toHaveBeenCalledTimes(1);
    });
});
