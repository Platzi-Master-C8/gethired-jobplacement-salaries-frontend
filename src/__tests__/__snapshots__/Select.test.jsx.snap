// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for component <Select /> Take Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth css-3lxuvf-MuiFormControl-root"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="false"
      id="label-select"
    >
      Select option
    </label>
    <div
      class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="label-select mui-component-select-select"
        class="MuiSelect-select MuiSelect-filled MuiFilledInput-input MuiInputBase-input css-d9oaum-MuiSelect-select-MuiInputBase-input-MuiFilledInput-input"
        id="mui-component-select-select"
        role="button"
        tabindex="0"
      >
        <span
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        name="select"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconFilled css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
